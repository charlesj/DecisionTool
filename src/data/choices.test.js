import { fromJS } from 'immutable'
import shortid from 'shortid'
import {
  addChoice,
  updateChoiceName,
  updateFactorScore,
  removeChoice,
} from './choices'

jest.mock('shortid')

const initState = fromJS({})

let id = 0

beforeEach(() => {
  shortid.mockImplementation(() => `id-${id++}`)
})

describe('addChoice', () => {

  it('adds the choice to the state', () => {
    expect(addChoice(initState)).toMatchSnapshot()
  })
})

describe('updateChoiceName', () => {
  it('updates the name of the choice given', () => {
    const choiceId = 'choice-id-1'
    const state = fromJS({
      choices: {
        [choiceId]: { name: 'old name' }
      }
    })

    expect(updateChoiceName(state, choiceId, 'new name')).toMatchSnapshot()
  })
})

describe('updateFactorScore', () => {
  it('updates the score for a given choice and factor', () => {

    const choiceId = 'choice-id-1'
    const state = fromJS({
      choices: {
        [choiceId]: { name: 'old name' }
      }
    })
    expect(updateFactorScore(state, choiceId, 'factor-id-1', 2)).toMatchSnapshot()
  })
})

describe('removeChoice', () => {
  it('removes the given choice', () => {
    const choiceId = 'choice-id-1'
    const state = fromJS({
      choices: {
        [choiceId]: { name: 'old name' },
        choice2: { name: 'this choice stays' }
      }
    })
    expect(removeChoice(state, choiceId)).toMatchSnapshot()
  })
})
