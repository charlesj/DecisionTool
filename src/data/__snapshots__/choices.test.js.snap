// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addChoice adds the choice to the state 1`] = `
Immutable.Map {
  "choices": Immutable.Map {
    "id-0": Immutable.Map {
      "name": "choice id-0",
    },
  },
}
`;

exports[`removeChoice removes the given choice 1`] = `
Immutable.Map {
  "choices": Immutable.Map {
    "choice2": Immutable.Map {
      "name": "this choice stays",
    },
  },
}
`;

exports[`updateChoiceName updates the name of the choice given 1`] = `
Immutable.Map {
  "choices": Immutable.Map {
    "choice-id-1": Immutable.Map {
      "name": "new name",
    },
  },
}
`;

exports[`updateFactorScore updates the score for a given choice and factor 1`] = `
Immutable.Map {
  "choices": Immutable.Map {
    "choice-id-1": Immutable.Map {
      "name": "old name",
      "scores": Immutable.Map {
        "factor-id-1": 2,
      },
    },
  },
}
`;
